### PR TITLE
Sockets fixes + testing

### LINKED ISSUE #7417: getaddrinfo and other blacklisted functions in EmScripten
Hello,

When trying to use EmScripten with version `1.38.12` (got unresolved symbol for `htons` in later version) `getaddrinfo` is not working properly.

I found out that `getaddrinfo` was part of the blacklisted functions in the libc, but the code is actually compiling and no warning / error is appearing.

Is that a bug ? (the code is working fine when the target is not wasm or asm.js) If not: 

* Do you suggest any work-around ? (HumbleNet seem to use `getaddrinfo` and working fine) ;
* It would be great to change the `system_libs.py` to actually throw an error at compiling when using a blacklisted functions. What do you think about that ?

### PR DESCRIPTION
**Pull request by kripken**
* Add some missing stuff in `deps_info.json` which fixes #7417.
* Move the non-browser-dependent parts of the sockets tests into other, so that they run on CI.

### CODE REVIEW

File: tests/test_other.py [hunk 1]
```diff
@@ -8837,3 +8837,225 @@ def test_emcc_parsing(self):
...
+  def test_getnameinfo(self):
+    self.do_run(open(path_from_root('tests', 'sockets', 'test_getnameinfo.c')).read(), 'success')
+
+  def test_gethostbyname(self):
+    self.do_run(open(path_from_root('tests', 'sockets', 'test_gethostbyname.c')).read(), 'success')
+
+  def test_getprotobyname(self):
+    self.do_run(open(path_from_root('tests', 'sockets', 'test_getprotobyname.c')).read(), 'success')
+
+  def test_link(self):
```
Comment (sbc100) [inline tests/test_other.py]: Better name for this test?

Comment (sbc100) [reply tests/test_other.py]: Why not put this in a file in tests/sockets like the above tests?

File: .circleci/config.yml [hunk 2]
```diff
@@ -211,7 +211,8 @@ jobs:
             EXTRA_AUTOSTART=$TMPDIR/autostart startx /usr/bin/openbox-session -- $DISPLAY -config ~/.config/X11/xorg.conf -nolisten tcp &
             cat $TMPDIR/fifo > /dev/null # wait until $EXTRA_AUTOSTART is spawned, which indicates the end of Openbox initialization
             rm -r $TMPDIR
-            EMCC_CORES=4 ./emscripten/tests/runner.py browser skip:browser.test_sdl2_mouse skip:browser.test_html5_webgl_create_context skip:browser.test_webgl_offscreen_canvas_in_pthread skip:browser.test_webgl_offscreen_canvas_in_mainthread_after_pthread skip:browser.test_glut_glutget
+            export EMCC_CORES=4
+            ./emscripten/tests/runner.py browser skip:browser.test_sdl2_mouse skip:browser.test_html5_webgl_create_context skip:browser.test_webgl_offscreen_canvas_in_pthread skip:browser.test_webgl_offscreen_canvas_in_mainthread_after_pthread skip:browser.test_glut_glutget
```
Comment (sbc100) [inline .circleci/config.yml]: Why this?

File: tests/test_other.py (hunk 1 above)
Comment (sbc100) [reply tests/test_other.py]: Oh I see you are just moving them.. not so important to fix that now then I guess.

File: .circleci/config.yml (hunk 2 above)
Comment (kripken) [reply .circleci/config.yml]: I just made it identical to the other place, as I noticed it was different (and this way seems nicer I think).

File: tests/test_other.py (hunk 1 above)
Comment (kripken) [reply tests/test_other.py]: Yeah, just moving. Although you made me think, if names collide here that would be bad... I verified there are no such cases though.

**Pull request merged**